asv_bench/benchmarks/benchmarks.py

```diff
@@ -359,3 +359,48 @@ def setup(self, shape, columns_number, ascending_list):
 
     def time_sort_values(self, shape, columns_number, ascending_list):
         execute(self.df.sort_values(self.columns, ascending=self.ascending))
+
+
+class BaseReadCsv:
+    # test data file can de created only once
+    def setup_cache(self, test_filename="io_test_file"):
+        test_filenames = {}
+        for shape in UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE]:
+            data_id = "_".join([str(size) for size in shape])
+            test_filenames[data_id] = f"{test_filename}_{data_id}.csv"
+            df = generate_dataframe(
+                ASV_USE_IMPL, "str_int", *shape, RAND_LOW, RAND_HIGH
+            )
+            df.to_csv(test_filenames[data_id], index=False)
+
+        return test_filenames
+
+    def setup(self, *args, **kwargs):
+        if ASV_USE_IMPL == "modin":
+            self.module = pd
+            # ray init
+            pd.DataFrame([])
+        else:
+            self.module = pandas
+
+class TimeReadCsv(BaseReadCsv):
+    param_names = ["shape"]
+    params = [UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE]]
+
+    def time_read_csv(self, test_filenames, shape):
+        data_id = "_".join([str(size) for size in shape])
+        execute(self.module.read_csv(test_filenames[data_id]))
+
+class TimeReadCsvSkiprows(BaseReadCsv):
+    param_names = ["shape", "skiprows"]
+    params = [
+        UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+        [
+            lambda x: x % 2,
+            np.arange(1, UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE][0][0] // 10),
+            np.arange(1, UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE][0][0], 2)],
+        ]
+
+    def time_skiprows(self, test_filenames, shape, skiprows):
```

Comment:
This function can be moved into `TimeReadCsv`

Reply:
Yes, it can, but in this case all parameters will be benchmarked together and parameters matrix can be big enough

Reply:
Yes, exactly in this case overall tests number will not change, so your suggestion was implemented.